salt/utils/process.py

```diff
@@ -288,7 +290,15 @@ class ThreadPool(object):
     '''
     def __init__(self,
                  num_threads=None,
-                 queue_size=0):
+                 queue_size=0,
+                 opts=None):
+        if not opts:
```

Comment:
I almost think we should make `opts` a required arg here. We should already have it generated by a caller anyway.

Reply:
Okay, I think it sounds long term'ish to get a nice consistent paradigm for stats. For now I will keep this discrete for simplicity. I should be able to push an update to this soon :-).

Reply:
OK, sounds good!